sec_certs/bsi_parse.py

```diff
@@ -16,115 +16,80 @@
-                a_link_list.append(a)
-        # Retrieve all the links from the anchor list in a proper format
-        for a in a_link_list:
-            self.link_list.append("https://www.bsi.bund.de/" + a['href'])
+        self.handler_list = \
+            [BsiHandler("https://www.bsi.bund.de/" + a['href'])
+             for a in self.soup.find_all('a', href=True, recursive=True)
+             if 'c-navigation-teaser' in str(a.get('class'))]
 
-        # Find the link to the archive pages
-        archive_link_list = []
-        for a in self.soup.find_all('a', href=True, recursive=True,
-                                    title=re.compile('Archive')):  # use of a regex to find the keyword
-            self.archive_list.append("https://www.bsi.bund.de/" + a['href'])
-
-
-"""
-Class used to browse BSI webpage -> get link to all categories
-This class will search for every category, and create a handler for each that will
-retrieve the links to the products
-"""
 
+class BsiHandler(BsiBrowser):
+    """
+    Class used to retrieve the links for all products under a category
+    """
 
-class BsiBrowser(BsiHandler):
-    handler_list: list[BsiHandler]
+    url: str
     link_list: list
+    soup: BeautifulSoup
 
     def __init__(self, url):
-        super().__init__(url)
+        self.url = url
+        self.soup = BeautifulSoup(requests.get(self.url).content, "html.parser")
+        self.link_list = []
         self.handler_list = []
 
     def parse(self):
-        """
-        Iterating over the anchors to filter them
-        """
-        a_link_list = []
-        for a in self.soup.find_all('a', href=True, recursive=True):
-            if 'c-navigation-teaser' in str(a.get('class')):
-                a_link_list.append(a)
-        """
-        Retrieve all the links from the anchor list in a proper format
-        """
-        for a in a_link_list:
-            link = "https://www.bsi.bund.de/" + a['href']
-            self.link_list.append(link)
-            self.handler_list.append(BsiHandler(link))
 
+        self.link_list = \
+            ["https://www.bsi.bund.de/" + a['href'] for a in self.soup.find_all('a', href=True, recursive=True) if
+             'white-space-nowrap' in str(a.parent.get('class'))]
 
-"""
-To end the processing, a last class will be used with final links
-to retrieve simple data written on the page
-"""
+        self.handler_list = \
+            [BsiHandler("https://www.bsi.bund.de/" + a['href'])
+             for a in self.soup.find_all('a', href=True, recursive=True, title=re.compile('Archive'))]
+
+        if len(self.handler_list) != 0:
+            for handler in self.handler_list:
+                handler.parse()
+
+        # ------------------------------------------------------------------------------------------------
+
+
+
+# To end the processing, a last class will be used with final links
+# to retrieve simple data written on the page
 
 
 class Bsitmp(ComplexSerializableType):
```

Comment:
Name of class is not really explanatory. Should be named something like `BSICertificate`